runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java

```diff
@@ -158,6 +158,18 @@ public long append(@Nonnull Set<UUID> streamIDs, @Nonnull Object object,
                 tokenResponse = new TokenResponse(
                         temp.getRespType(), tokenResponse.getConflictKey(),
                         temp.getToken(), temp.getBackpointerMap());
+
+            } catch (StaleTokenException se) {
+                // the epoch changed from when we grabbed the token from sequencer
+                log.warn("StaleToken[{}]: streams {}", tokenResponse.getTokenValue(),
+                        streamIDs.stream().map(Utils::toReadableId).collect(Collectors.toSet()));
+
+                throw new TransactionAbortedException(
+                        conflictInfo,
+                        tokenResponse.getConflictKey(),
+                        AbortCause.NEW_SEQUENCER, // TODO define a new AbortCause?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)